### PR TITLE
From db snapshot

### PR DESCRIPTION
**Pull request by pepoviola**
cc @michalkucharczyk 

- Add `db_snapshot` / `default_db_snapshot ` config (nodes/relay-chain)
- Add test using db_snapshot.

### CODE REVIEW

**Comment by michalkucharczyk** [comment]
What would be the usage of  `db_snapshot`?

Like this?
```
[relaychain]
default_image = "docker.io/paritypr/polkadot-debug:master"
default_command = "polkadot"
default_args = [ "-lparachain=debug" ]

chain = "rococo-local"

  [[relaychain.nodes]]
  name = "alice"
  validator = true
  db_snapshot = "https://storage.googleapis.com/zombienet-db-snaps/chains-test-alice.tgz"
  
  [[relaychain.nodes]]
  name = "bob"
  validator = true
  db_snapshot = "https://storage.googleapis.com/zombienet-db-snaps/chains-test-bob.tgz"
```

Maybe it would be good to include some examples, and also add a note about this feature do documentation?

**Comment by pepoviola** [comment]
> What would be the usage of `db_snapshot`?
> 
> Like this?
> 
> ```
> [relaychain]
> default_image = "docker.io/paritypr/polkadot-debug:master"
> default_command = "polkadot"
> default_args = [ "-lparachain=debug" ]
> 
> chain = "rococo-local"
> 
>   [[relaychain.nodes]]
>   name = "alice"
>   validator = true
>   db_snapshot = "https://storage.googleapis.com/zombienet-db-snaps/chains-test-alice.tgz"
>   
>   [[relaychain.nodes]]
>   name = "bob"
>   validator = true
>   db_snapshot = "https://storage.googleapis.com/zombienet-db-snaps/chains-test-bob.tgz"
> ```
> 
> Maybe it would be good to include some examples, and also add a note about this feature do documentation?

Yes, you can also use like this 
```
[relaychain]
default_image = "docker.io/paritypr/polkadot-debug:master"
default_command = "polkadot"
default_args = [ "-lparachain=debug" ]
default_db_snapshot = "https://storage.googleapis.com/zombienet-db-snaps/chains-test.tgz"

chain = "rococo-local"

  [[relaychain.nodes]]
  name = "alice"
  validator = true

  [[relaychain.nodes]]
  name = "bob"
  validator = true
 ```
 
 I will update the docs in a follow up pr.

File: tests/0013-db-snapshot.zndsl [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+Network: ./0013-db-snapshot.toml
+Creds: config
+
+
+# well known functions
+alice: is up
+bob: is up
+
+# check balance
+alice: js-script ./check-balance.js with "alice" return is 900000999940423862 within 200 seconds
```
Comment (michalkucharczyk) [inline tests/0013-db-snapshot.zndsl:11]: nit: we could also check that block height is equal to the height from db.

**Review by michalkucharczyk — APPROVED** [review]

File: javascript/packages/orchestrator/src/providers/podman/podmanClient.ts [hunk 2]
```diff
@@ -360,12 +362,25 @@ export class PodmanClient extends Client {
...
+    if (dbSnapshot) {
+      // we need to get the snapshot from a public access
+      // and extract to /data
+      await makeDir(`${dataPath.hostPath.path}/chains`, true);
+
+      await downloadFile(dbSnapshot, `${dataPath.hostPath.path}/chains/db.tgz`);
+      await this.runCommand([
+        "-c",
+        `cd ${dataPath.hostPath.path}/chains && tar -xzvf db.tgz`,
+      ]);
```
Comment (michalkucharczyk) [inline javascript/packages/orchestrator/src/providers/podman/podmanClient.ts:380]:
```suggestion
      const args = [ "-c", `cd ${dataPath.hostPath.path}/chains && tar -xzvf db.tgz`, ];
      const result = await execa("bash", args);

      if (result.exitCode != 0) {
        throw new Error(`Command failed (${args}) : ${result}`);
      };
```

**Review by michalkucharczyk — CHANGES_REQUESTED** [review]
fails for `podman` provider.

**Comment by michalkucharczyk** [comment]
> > fails for `podman` provider.
> 
> Can you share the error please?

```
Error: Command failed with exit code 125: podman -c cd /tmp/zombie-6fb1cc5d_-1858348-hcgAgBX105OF/alice/data/chains && tar -xzvf db.tgz
Error: missing command 'podman COMMAND'
Try 'podman --help' for more information.
    at makeError (/snapshot/javascript/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/snapshot/javascript/node_modules/execa/index.js:118:26)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  shortMessage: 'Command failed with exit code 125: podman -c cd /tmp/zombie-6fb1cc5d_-1858348-hcgAgBX105OF/alice/data/chains && tar -xzvf db.tgz',
  command: 'podman -c cd /tmp/zombie-6fb1cc5d_-1858348-hcgAgBX105OF/alice/data/chains && tar -xzvf db.tgz',
  escapedCommand: 'podman -c "cd /tmp/zombie-6fb1cc5d_-1858348-hcgAgBX105OF/alice/data/chains && tar -xzvf db.tgz"',
  exitCode: 125,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: "Error: missing command 'podman COMMAND'\n" +
    "Try 'podman --help' for more information.",
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
Error: Command failed with exit code 125: podman -c cd /tmp/zombie-6fb1cc5d_-1858348-hcgAgBX105OF/alice/data/chains && tar -xzvf db.tgz
Error: missing command 'podman COMMAND'
Try 'podman --help' for more information.
    at makeError (/snapshot/javascript/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/snapshot/javascript/node_modules/execa/index.js:118:26)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  shortMessage: 'Command failed with exit code 125: podman -c cd /tmp/zombie-6fb1cc5d_-1858348-hcgAgBX105OF/alice/data/chains && tar -xzvf db.tgz',
  command: 'podman -c cd /tmp/zombie-6fb1cc5d_-1858348-hcgAgBX105OF/alice/data/chains && tar -xzvf db.tgz',
  escapedCommand: 'podman -c "cd /tmp/zombie-6fb1cc5d_-1858348-hcgAgBX105OF/alice/data/chains && tar -xzvf db.tgz"',
  exitCode: 125,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: "Error: missing command 'podman COMMAND'\n" +
    "Try 'podman --help' for more information.",
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
```

**Comment by pepoviola** [comment]
@michalkucharczyk podman fixed

image

**Pull request merged**